src/test/scala/org/broadinstitute/dsde/rawls/expressions/SlickSimpleExpressionParserTest.scala

```diff
@@ -0,0 +1,230 @@
+    }
+  }
+
+  test("reserved attribute expression") {
+    withTestWorkspace { workspaceContext =>
+      val evaluator = expressionEvaluator
+      assertResult(Seq(AttributeString("sample1"))) {
+        runAndWait(evaluator.evalFinalAttribute(workspaceContext, "Sample", "sample1", "this.name").get)
+      }
+
+      assertResult(Seq(AttributeString("Sample"), AttributeString("Sample"), AttributeString("Sample"))) {
+        runAndWait(evaluator.evalFinalAttribute(workspaceContext, "SampleSet", "sset1", "this.samples.entityType").get)
+      }
+
+      assertResult(Seq(AttributeString(workspaceContext.workspace.name))) {
+        runAndWait(evaluator.evalFinalAttribute(workspaceContext, "SampleSet", "sset1", "workspace.name").get)
+      }
+
+      intercept[RawlsException] {
+        runAndWait(evaluator.evalFinalAttribute(workspaceContext, "SampleSet", "sset1", "workspace.entityType").get)
+      }
+    }
+  }
+
+  test("workspace attribute expression") {
+    withTestWorkspace { workspaceContext =>
+      val evaluator = expressionEvaluator
+
+      assertResult(Seq(testData.wsAttrs.get("string").get)) {
+        runAndWait(evaluator.evalFinalAttribute(workspaceContext, "dummy text", "dummy text", "workspace.string").get)
+      }
+
+      assertResult(Seq(testData.wsAttrs.get("number").get)) {
+        runAndWait(evaluator.evalFinalAttribute(workspaceContext, "dummy text", "dummy text", "workspace.number").get)
+      }
+
+      assertResult(Seq(testData.sample1.attributes.get("type").get)) {
+        val attributesPlusReference = testData.workspace.attributes + ("sample1ref" -> AttributeEntityReference("Sample", "sample1"))
+        runAndWait(workspaceQuery.save(testData.workspace.copy(attributes = attributesPlusReference)))
+
+        runAndWait(evaluator.evalFinalAttribute(workspaceContext, "dummy text", "dummy text", "workspace.sample1ref.type").get)
+      }
+
+      assertResult(Seq()) {
+        runAndWait(evaluator.evalFinalAttribute(workspaceContext, "dummy text", "dummy text", "workspace.empty").get)
+      }
+
+      assertResult(Seq(AttributeString("another string"), AttributeString("true"))) {
+        runAndWait(evaluator.evalFinalAttribute(workspaceContext, "dummy text", "dummy text", "workspace.values").get)
+      }
+
+      assertResult(Seq()) {
+        runAndWait(evaluator.evalFinalAttribute(workspaceContext, "dummy text", "dummy text", "workspace.this_attribute_is_not_present").get)
+      }
+
+      assert {
+        evaluator.evalFinalAttribute(workspaceContext, "dummy text", "dummy text", "workspace").isFailure
+      }
+
+      assert {
+        evaluator.evalFinalAttribute(workspaceContext, "dummy text", "dummy text", "workspace.").isFailure
+      }
+
+// https://broadinstitute.atlassian.net/browse/GAWB-349 figure out how to add this back in one day
+//      assert {
+//        evaluator.evalFinalAttribute(workspaceContext, "dummy text", "dummy text", "workspace.missing.also_missing").isFailure
+//      }
```

Comment:
note that I could not figure out how to do this short of a separate query for each level of the expression